pkg/clusters/types/eks/aws-operations/eks.go

```diff
@@ -0,0 +1,479 @@
+package aws_operations
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/pkg/errors"
+	"github.com/samber/lo"
+	"github.com/weaveworks/eksctl/pkg/ami"
+	eksctlapi "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/authconfigmap"
+	eksiam "github.com/weaveworks/eksctl/pkg/iam"
+	"github.com/weaveworks/eksctl/pkg/nodebootstrap"
+	"k8s.io/client-go/kubernetes"
+	"os"
+	"time"
+)
+
+const (
+	DefaultNodeGroupName     = "default-node-group"
+	DefaultKubernetesSvcCIDR = "172.20.0.0/16"
+	kubernetesTagFormat      = "kubernetes.io/cluster/%s"
+	envKeyNodeSSHKeyName     = "EKS_NODE_SSH_KEY"
+
+	TagNameCreateBy = "ktf_created_by"
+)
+
+func CreateEKSClusterAll(ctx context.Context, cfg aws.Config, clusterName,
```

Comment:
It looks like to be a function to create a cluster and all its dependencies. Could we add comments to tell this?